packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect, useCallback, useState } from "react";
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckIcon,
+  CloseIcon,
+  Flex,
+  FormControl,
+  HStack,
+  IconButton,
+  InfoOutlineIcon,
+  Popover,
+  Spinner,
+  Stack,
+  Switch,
+  Text,
+  WarningOutlineIcon
+} from "native-base";
+import { TransactionStatus } from "@usedapp/core";
+import { TokenInput } from "../../core";
+import { AddressInput, isAddressValid } from "../../core/inputs/AddressInput";
+import { BigNumber } from "ethers";
+import { ExplorerLink } from "../../core/web3/ExplorerLink";
+
+type OnBridge = (amount: string, sourceChain: string, target?: string) => Promise<void>;
+
+const Status = ({ result, ...props }: { result?: string }) => {
+  switch (result) {
+    case undefined:
+    case "Mining":
+    case "PendingSignature":
+    default:
+      return <Spinner {...props} />;
+    case "Success":
+      return <CheckIcon size="5" mt="0.5" color="emerald.500" {...props} />;
+    case "Fail":
+    case "Exception":
+      return <CloseIcon size="5" mt="0.5" color="danger.500" {...props} />;
+  }
+};
+
+const StatusBox = ({
+  txStatus,
+  text,
+  infoText,
+  sourceChain
+}: {
+  txStatus?: Partial<TransactionStatus>;
+  text: string;
+  infoText?: string;
+  sourceChain: "fuse" | "celo";
+}) => (
+  <Stack mt="2" alignItems="center" direction={["column", "column", "row"]}>
+    <HStack alignItems="center" flex="2 0">
+      <Box>
+        <Status result={txStatus?.status} />
+      </Box>
+      <Box flex={"1 1"}>
+        <Text color="emerald.500" fontSize="md" ml="2">
+          {text}
+        </Text>
+      </Box>
+      {infoText && (
+        <Popover
+          trigger={triggerProps => {
+            return <IconButton {...triggerProps} icon={<InfoOutlineIcon />} />;
+          }}
+        >
+          <Popover.Content accessibilityLabel={infoText} w="md">
+            <Popover.CloseButton />
+            <Popover.Body>{infoText}</Popover.Body>
+          </Popover.Content>
+        </Popover>
+      )}
+    </HStack>
+    <Flex flex="1 1" alignItems="center" maxWidth="100%">
+      {txStatus && <ExplorerLink chainId={txStatus.chainId} addressOrTx={txStatus.transaction?.hash} />}
+    </Flex>
+  </Stack>
+);
+
+export const MicroBridge = ({
+  useBalanceHook,
```

Comment:
passing hooks as props is a bad design because props could be changed
hook function itself should be immutable

1. move hooks to some third file (to avoid circular imports)
2. import them here
3. remove from props, use from import

Reply:
the idea is that this would be a generic component so, someone else can pass any hook implementation it wants.
there's not reason why hooks cant be passed down